src/zope/testrunner/formatter.py

```diff
@@ -439,16 +428,15 @@ def tigetnum(attr, default=None):
     try:
         import curses
     except ImportError:
-        # avoid reimporting a broken module in python 2.3
+        # avoid reimporting a broken module
         sys.modules['curses'] = None
     else:
         # If sys.stdout is not a real file object (e.g. in unit tests that
         # use various wrappers), you get an error, different depending on
         # Python version:
         expected_exceptions = (curses.error, TypeError, AttributeError)
-        if sys.version_info >= (3,):
-            import io
-            expected_exceptions += (io.UnsupportedOperation, )
+        import io
+        expected_exceptions += (io.UnsupportedOperation, )
```

Comment:
You might want to do on the lines below:

```python
try:
    ...
except (curses.error, TypeError, AttributeError, io.UnsupportedOperation)
```

Reply:
Implemented in https://github.com/zopefoundation/zope.testrunner/pull/146/commits/a5a900eda3808ad2a181650c6f741afb9b6f5147.